server/tests/cluster.rs

```diff
@@ -0,0 +1,104 @@
+use std::time::Duration;
+
+use enumset::enum_set;
+use futures_util::StreamExt;
+use restate_local_cluster_runner::{
+    cluster::Cluster,
+    node::{BinarySource, Node},
+};
+use restate_types::{
+    config::Configuration,
+    nodes_config::Role,
+    PlainNodeId,
+};
+
+#[tokio::test]
+async fn node_id_mismatch() {
+    let base_config = Configuration::default();
+    let base_dir = tempfile::tempdir().expect("tempdir failed").into_path();
```

Comment:
Can be an improvement in a future PR: One thing that can be very helpful is to allow the temp base-dir to stay (with its path logged/printed after a test run based on an environment variable (see bifrost-benchpress for an example).